diff --git a/bin/jbuild b/bin/jbuild
index ddafe43..528a538 100644
--- a/bin/jbuild
+++ b/bin/jbuild
@@ -12,7 +12,7 @@
   (flags (:standard -safe-string))
   (libraries (systemd
               xapi-types
-              xcp.v6)
+              xapi-idl.v6)
   )
  )
 )
diff --git a/bin/v6_server.ml b/bin/v6_server.ml
index 582e945..4fb8313 100644
--- a/bin/v6_server.ml
+++ b/bin/v6_server.ml
@@ -1,4 +1,4 @@
-module S = V6_interface.RPC_API(Idl.GenServerExn ())
+module S = V6_interface.RPC_API(Idl.Exn.GenServer ())
 
 type context = unit
 
diff --git a/bin/xcp_featured.ml b/bin/xcp_featured.ml
index d6fc000..bead50f 100644
--- a/bin/xcp_featured.ml
+++ b/bin/xcp_featured.ml
@@ -23,7 +23,7 @@ let () =
   let server = Xcp_service.make
     ~path:!V6_interface.default_path
     ~queue_name:!V6_interface.queue_name
-    ~rpc_fn:(Idl.server V6_server.S.implementation)
+    ~rpc_fn:(Idl.Exn.server V6_server.S.implementation)
     ()
   in
 
